utils/heap/queue.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package heap
+
+import (
+	stdheap "container/heap"
+
+	"github.com/ava-labs/avalanchego/utils"
+)
+
+var _ stdheap.Interface = (*queue[int])(nil)
+
+// NewQueue returns an empty min-heap ordered by less. See QueueOf for more.
```

Comment:
This comment seems out of date (or we need a comment on `QueueOf`)

Reply:
Updated